kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemManagerDbusWrapper.java

```diff
@@ -272,4 +284,84 @@ protected void resetHandlersDisable(String deviceId) {
             }
         }
     }
+
+    protected void failedResetHandlerEnable(String deviceId, Optional<String> modemManagerDbusPath, int delayMinutes)
+            throws DBusException {
+        if (!modemManagerDbusPath.isPresent()) {
+            logger.warn("Cannot retrieve modem device for {}. Skipping modem reset monitor setup.", deviceId);
+            return;
+        }
+
+        Modem mmModemDevice = this.dbusConnection.getRemoteObject(MM_BUS_NAME, modemManagerDbusPath.get(), Modem.class);
+
+        MMFailedModemResetHandler resetHandler = new MMFailedModemResetHandler(mmModemDevice,
+                modemManagerDbusPath.get(), delayMinutes);
+        resetHandler.schedule();
+
+        this.failedModemHandlers.put(deviceId, resetHandler);
+    }
+
+    protected void failedResetHandlersDisable() {
+        for (String deviceId : this.failedModemHandlers.keySet()) {
+            failedResetHandlersDisable(deviceId);
+        }
+        this.modemHandlers.clear();
+    }
+
+    protected void failedResetHandlersDisable(String deviceId) {
+        if (this.failedModemHandlers.containsKey(deviceId)) {
+            MMFailedModemResetHandler handler = this.failedModemHandlers.get(deviceId);
+            handler.cancel();
+        }
+    }
+
+    protected boolean isMMFailedModemResetTimerArmed(String deviceId) {
+        return this.failedModemHandlers.containsKey(deviceId);
+    }
+
+    private class MMFailedModemResetTimerTask extends NMModemResetTimerTask {
+
+        private final String modemManagerDbusPath;
+
+        public MMFailedModemResetTimerTask(Modem modem, String modemManagerDbusPath) {
+            super(modem);
+            this.modemManagerDbusPath = modemManagerDbusPath;
+        }
+
+        @Override
+        public void run() {
+            try {
+                MMModemState modemState = getMMModemState(modemManagerDbusPath);
+                if (MMModemState.MM_MODEM_STATE_FAILED.equals(modemState)) {
+                    super.run();
+                }
+            } catch (DBusException e) {
+                ModemManagerDbusWrapper.logger.warn("Couldn't get state of modem interface, caused by:", e);
```

Comment:
I don't super like this. Why not use the logger within the super class? See:  https://github.com/eclipse-kura/kura/blob/e6c27549de30c95e2c3926e0cecbf7c9253c52ba/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/signal/handlers/NMModemResetTimerTask.java#L25

Reply:
I think that I cannot use it since it is private.

Reply:
Sorry, I implicitly assumed you would need to change it to `protected`. I think the change in visibility might make sense here. Let me know what you think.